### PR TITLE
test: Verify ServiceLoadBalancer in e2e Docker and Nutanix tests

### PR DESCRIPTION
**Pull request by dlipovetsky**
**What problem does this PR solve?**:
Verifies that the ServiceLoadBalancer provider MetalLB is deployed, and its configuration created, in Docker and Nutanix clusters.
 
**Which issue(s) this PR fixes**:
Fixes #

**How Has This Been Tested?**:

**Special notes for your reviewer**:

### CODE REVIEW

File: hack/examples/patches/metallb.yaml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# Copyright 2024 Nutanix. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+- op: "add"
+  path: "/spec/topology/variables/0/value/addons/serviceLoadBalancer"
+  value:
+    provider: MetalLB
+    configuration:
+      addressRanges:
+        - start: 198.18.1.1
```
Comment (faiq) [inline hack/examples/patches/metallb.yaml:10]: i see that you have these in an order that makes sense here, but kustomize reverses the order. a little frustrating, but thanks for trying to make it more readable.

Comment (dlipovetsky) [reply hack/examples/patches/metallb.yaml:10]: Yeah, I noticed that, too.

 I started off by modifying the cluster template directly. I didn't know that we generated the example cluster templates from `hack/examples`. Now I know, and I'll document it in a separate PR :smile:

File: pkg/handlers/generic/lifecycle/serviceloadbalancer/metallb/configuration.go [hunk 2]
```diff
@@ -60,7 +60,7 @@ func configurationObjects(input *configurationInput) ([]unstructured.Unstructure
 		return nil, fmt.Errorf("failed to set L2Advertisement .spec.ipAddressPools: %w", err)
 	}
 
-	return []unstructured.Unstructured{
+	return []*unstructured.Unstructured{
```
Comment (dkoshkin) [inline pkg/handlers/generic/lifecycle/serviceloadbalancer/metallb/configuration.go:63]: Goog change 🙏

**Review by dkoshkin — APPROVED** [review]
🤔 how do we know that any of these checks are actually running, the runner really doesn't show anything https://github.com/nutanix-cloud-native/cluster-api-runtime-extensions-nutanix/actions/runs/9812003669/job/27095240802?pr=788#step:9:176

**Comment by dlipovetsky** [comment]
> 🤔 how do we know that any of these checks are actually running, the runner really doesn't show anything https://github.com/nutanix-cloud-native/cluster-api-runtime-extensions-nutanix/actions/runs/9812003669/job/27095240802?pr=788#step:9:176

That e2e test uses the ClusterResourceSet strategy. At this time, ServiceLoadBalancer providers are only installed using the Helm strategy. Therefore, in this test, the check is skipped.

**Comment by dlipovetsky** [comment]
Rebased on main after #791 merged.

**Comment by dkoshkin** [comment]
> > 🤔 how do we know that any of these checks are actually running, the runner really doesn't show anything https://github.com/nutanix-cloud-native/cluster-api-runtime-extensions-nutanix/actions/runs/9812003669/job/27095240802?pr=788#step:9:176
> 
> That e2e test uses the ClusterResourceSet strategy. At this time, ServiceLoadBalancer providers are only installed using the Helm strategy. Therefore, in this test, the check is skipped.

Sorry I wasn't clear, I meant that the logs don't have anything about any of these checks (not specifically this new one)

**Comment by dlipovetsky** [comment]
> > > 🤔 how do we know that any of these checks are actually running, the runner really doesn't show anything https://github.com/nutanix-cloud-native/cluster-api-runtime-extensions-nutanix/actions/runs/9812003669/job/27095240802?pr=788#step:9:176
> > 
> > 
> > That e2e test uses the ClusterResourceSet strategy. At this time, ServiceLoadBalancer providers are only installed using the Helm strategy. Therefore, in this test, the check is skipped.
> 
> Sorry I wasn't clear, I meant that the logs don't have anything about any of these checks (not specifically this new one)

Ahh, ok. I was aiming for parity with other checks; I'll see what I've missed and update the PR.

**Comment by dlipovetsky** [comment]
We're not seeing verbose output. Historically, we enabled verbose output if the `CI` make variable was defined. It looks like we've never defined it, though.

**Comment by dlipovetsky** [comment]
Going to rebase on #797 so we can see the output of the new checks

**Comment by dlipovetsky** [comment]
Verified new step in the log:
https://github.com/nutanix-cloud-native/cluster-api-runtime-extensions-nutanix/actions/runs/9845606994/job/27181724457?pr=788#step:9:478

**Review by faiq — APPROVED** [review]

**Pull request merged**